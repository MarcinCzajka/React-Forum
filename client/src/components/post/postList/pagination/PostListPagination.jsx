import React from 'react';
import PropTypes from 'prop-types';
import { Grid, Pagination } from "semantic-ui-react";

const PostListPagination = props => {
    if(props.totalPages > 1) return (
        <Grid.Row columns={3}>
            <Grid.Column floated={ (props.windowWidth < 960 ? 'left' : 'right') }>
                <Pagination onPageChange={props.changePage} activePage={props.activePage} totalPages={props.totalPages} />
            </Grid.Column>
        </Grid.Row>
    );

    return ''
}

PostListPagination.defaultProps = {
    activePage: 1,
    totalPages: 1
}

PostListPagination.propTypes = {
    activePage: PropTypes.number,
    totalPages: PropTypes.number,
    windowWidth: PropTypes.number,
    changePage: PropTypes.func.isRequired
}

export default PostListPagination