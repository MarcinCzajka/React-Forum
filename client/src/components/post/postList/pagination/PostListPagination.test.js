import React from 'react';
import { shallow } from 'enzyme';
import { create } from "react-test-renderer";
import PostListPagination from './PostListPagination';

describe('PostListPagination component', () => {
    test('renders without crashing', () => {
        shallow(<PostListPagination changePage={()=>{}} />);
    });

    test('match snapshot w/o props', () => {
        const component = create(<PostListPagination changePage={()=>{}} />);
        expect(component).toMatchSnapshot();
    });

    test('match snapshot wide screen /w props 1/3', () => {
        const component = create(<PostListPagination 
            activePage={1}
            totalPages={5}
            windowWidth={1024}
            changePage={()=>{}} 
        />);

        expect(component).toMatchSnapshot();
    });

    test('match snapshot rwd /w props 1/3', () => {
        const component = create(<PostListPagination 
            activePage={1}
            totalPages={5}
            windowWidth={480}
            changePage={()=>{}} 
        />);
        
        expect(component).toMatchSnapshot();
    });

    test('match snapshot wide screen /w props 2/3', () => {
        const component = create(<PostListPagination 
            activePage={3}
            totalPages={5}
            windowWidth={1024}
            changePage={()=>{}} 
        />);

        expect(component).toMatchSnapshot();
    });

    test('match snapshot rwd /w props 2/3', () => {
        const component = create(<PostListPagination 
            activePage={3}
            totalPages={5}
            windowWidth={480}
            changePage={()=>{}} 
        />);
        
        expect(component).toMatchSnapshot();
    });

    test('match snapshot wide screen /w props 3/3', () => {
        const component = create(<PostListPagination 
            activePage={10}
            totalPages={10}
            windowWidth={1024}
            changePage={()=>{}} 
        />);

        expect(component).toMatchSnapshot();
    });

    test('match snapshot rwd /w props 3/3', () => {
        const component = create(<PostListPagination 
            activePage={10}
            totalPages={10}
            windowWidth={480}
            changePage={()=>{}} 
        />);
        
        expect(component).toMatchSnapshot();
    });
})