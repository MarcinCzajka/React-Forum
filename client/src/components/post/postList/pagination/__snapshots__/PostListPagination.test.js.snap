// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PostListPagination component match snapshot rwd /w props 1/3 1`] = `
<div
  className="three column row"
>
  <div
    className="left floated column"
  >
    <div
      aria-label="Pagination Navigation"
      className="ui pagination menu"
      role="navigation"
    >
      <a
        aria-current={false}
        aria-disabled={false}
        aria-label="First item"
        className="item"
        onClick={[Function]}
        onKeyDown={[Function]}
        tabIndex={0}
        type="firstItem"
        value={1}
      >
        «
      </a>
      <a
        aria-current={false}
        aria-disabled={false}
        aria-label="Previous item"
        className="item"
        onClick={[Function]}
        onKeyDown={[Function]}
        tabIndex={0}
        type="prevItem"
        value={1}
      >
        ⟨
      </a>
      <a
        aria-current={true}
        aria-disabled={false}
        className="active item"
        onClick={[Function]}
        onKeyDown={[Function]}
        tabIndex={0}
        type="pageItem"
        value={1}
      >
        1
      </a>
      <a
        aria-current={false}
        aria-disabled={false}
        className="item"
        onClick={[Function]}
        onKeyDown={[Function]}
        tabIndex={0}
        type="pageItem"
        value={2}
      >
        2
      </a>
      <a
        aria-current={false}
        aria-disabled={false}
        className="item"
        onClick={[Function]}
        onKeyDown={[Function]}
        tabIndex={0}
        type="pageItem"
        value={3}
      >
        3
      </a>
      <a
        aria-current={false}
        aria-disabled={false}
        className="item"
        onClick={[Function]}
        onKeyDown={[Function]}
        tabIndex={0}
        type="pageItem"
        value={4}
      >
        4
      </a>
      <a
        aria-current={false}
        aria-disabled={false}
        className="item"
        onClick={[Function]}
        onKeyDown={[Function]}
        tabIndex={0}
        type="pageItem"
        value={5}
      >
        5
      </a>
      <a
        aria-current={false}
        aria-disabled={false}
        aria-label="Next item"
        className="item"
        onClick={[Function]}
        onKeyDown={[Function]}
        tabIndex={0}
        type="nextItem"
        value={2}
      >
        ⟩
      </a>
      <a
        aria-current={false}
        aria-disabled={false}
        aria-label="Last item"
        className="item"
        onClick={[Function]}
        onKeyDown={[Function]}
        tabIndex={0}
        type="lastItem"
        value={5}
      >
        »
      </a>
    </div>
  </div>
</div>
`;

exports[`PostListPagination component match snapshot rwd /w props 2/3 1`] = `
<div
  className="three column row"
>
  <div
    className="left floated column"
  >
    <div
      aria-label="Pagination Navigation"
      className="ui pagination menu"
      role="navigation"
    >
      <a
        aria-current={false}
        aria-disabled={false}
        aria-label="First item"
        className="item"
        onClick={[Function]}
        onKeyDown={[Function]}
        tabIndex={0}
        type="firstItem"
        value={1}
      >
        «
      </a>
      <a
        aria-current={false}
        aria-disabled={false}
        aria-label="Previous item"
        className="item"
        onClick={[Function]}
        onKeyDown={[Function]}
        tabIndex={0}
        type="prevItem"
        value={2}
      >
        ⟨
      </a>
      <a
        aria-current={false}
        aria-disabled={false}
        className="item"
        onClick={[Function]}
        onKeyDown={[Function]}
        tabIndex={0}
        type="pageItem"
        value={1}
      >
        1
      </a>
      <a
        aria-current={false}
        aria-disabled={false}
        className="item"
        onClick={[Function]}
        onKeyDown={[Function]}
        tabIndex={0}
        type="pageItem"
        value={2}
      >
        2
      </a>
      <a
        aria-current={true}
        aria-disabled={false}
        className="active item"
        onClick={[Function]}
        onKeyDown={[Function]}
        tabIndex={0}
        type="pageItem"
        value={3}
      >
        3
      </a>
      <a
        aria-current={false}
        aria-disabled={false}
        className="item"
        onClick={[Function]}
        onKeyDown={[Function]}
        tabIndex={0}
        type="pageItem"
        value={4}
      >
        4
      </a>
      <a
        aria-current={false}
        aria-disabled={false}
        className="item"
        onClick={[Function]}
        onKeyDown={[Function]}
        tabIndex={0}
        type="pageItem"
        value={5}
      >
        5
      </a>
      <a
        aria-current={false}
        aria-disabled={false}
        aria-label="Next item"
        className="item"
        onClick={[Function]}
        onKeyDown={[Function]}
        tabIndex={0}
        type="nextItem"
        value={4}
      >
        ⟩
      </a>
      <a
        aria-current={false}
        aria-disabled={false}
        aria-label="Last item"
        className="item"
        onClick={[Function]}
        onKeyDown={[Function]}
        tabIndex={0}
        type="lastItem"
        value={5}
      >
        »
      </a>
    </div>
  </div>
</div>
`;

exports[`PostListPagination component match snapshot rwd /w props 3/3 1`] = `
<div
  className="three column row"
>
  <div
    className="left floated column"
  >
    <div
      aria-label="Pagination Navigation"
      className="ui pagination menu"
      role="navigation"
    >
      <a
        aria-current={false}
        aria-disabled={false}
        aria-label="First item"
        className="item"
        onClick={[Function]}
        onKeyDown={[Function]}
        tabIndex={0}
        type="firstItem"
        value={1}
      >
        «
      </a>
      <a
        aria-current={false}
        aria-disabled={false}
        aria-label="Previous item"
        className="item"
        onClick={[Function]}
        onKeyDown={[Function]}
        tabIndex={0}
        type="prevItem"
        value={9}
      >
        ⟨
      </a>
      <a
        aria-current={false}
        aria-disabled={false}
        className="item"
        onClick={[Function]}
        onKeyDown={[Function]}
        tabIndex={0}
        type="pageItem"
        value={1}
      >
        1
      </a>
      <a
        aria-current={false}
        aria-disabled={true}
        className="item"
        onClick={[Function]}
        onKeyDown={[Function]}
        tabIndex={-1}
        type="ellipsisItem"
        value={5}
      >
        ...
      </a>
      <a
        aria-current={false}
        aria-disabled={false}
        className="item"
        onClick={[Function]}
        onKeyDown={[Function]}
        tabIndex={0}
        type="pageItem"
        value={6}
      >
        6
      </a>
      <a
        aria-current={false}
        aria-disabled={false}
        className="item"
        onClick={[Function]}
        onKeyDown={[Function]}
        tabIndex={0}
        type="pageItem"
        value={7}
      >
        7
      </a>
      <a
        aria-current={false}
        aria-disabled={false}
        className="item"
        onClick={[Function]}
        onKeyDown={[Function]}
        tabIndex={0}
        type="pageItem"
        value={8}
      >
        8
      </a>
      <a
        aria-current={false}
        aria-disabled={false}
        className="item"
        onClick={[Function]}
        onKeyDown={[Function]}
        tabIndex={0}
        type="pageItem"
        value={9}
      >
        9
      </a>
      <a
        aria-current={true}
        aria-disabled={false}
        className="active item"
        onClick={[Function]}
        onKeyDown={[Function]}
        tabIndex={0}
        type="pageItem"
        value={10}
      >
        10
      </a>
      <a
        aria-current={false}
        aria-disabled={false}
        aria-label="Next item"
        className="item"
        onClick={[Function]}
        onKeyDown={[Function]}
        tabIndex={0}
        type="nextItem"
        value={10}
      >
        ⟩
      </a>
      <a
        aria-current={false}
        aria-disabled={false}
        aria-label="Last item"
        className="item"
        onClick={[Function]}
        onKeyDown={[Function]}
        tabIndex={0}
        type="lastItem"
        value={10}
      >
        »
      </a>
    </div>
  </div>
</div>
`;

exports[`PostListPagination component match snapshot w/o props 1`] = `""`;

exports[`PostListPagination component match snapshot wide screen /w props 1/3 1`] = `
<div
  className="three column row"
>
  <div
    className="right floated column"
  >
    <div
      aria-label="Pagination Navigation"
      className="ui pagination menu"
      role="navigation"
    >
      <a
        aria-current={false}
        aria-disabled={false}
        aria-label="First item"
        className="item"
        onClick={[Function]}
        onKeyDown={[Function]}
        tabIndex={0}
        type="firstItem"
        value={1}
      >
        «
      </a>
      <a
        aria-current={false}
        aria-disabled={false}
        aria-label="Previous item"
        className="item"
        onClick={[Function]}
        onKeyDown={[Function]}
        tabIndex={0}
        type="prevItem"
        value={1}
      >
        ⟨
      </a>
      <a
        aria-current={true}
        aria-disabled={false}
        className="active item"
        onClick={[Function]}
        onKeyDown={[Function]}
        tabIndex={0}
        type="pageItem"
        value={1}
      >
        1
      </a>
      <a
        aria-current={false}
        aria-disabled={false}
        className="item"
        onClick={[Function]}
        onKeyDown={[Function]}
        tabIndex={0}
        type="pageItem"
        value={2}
      >
        2
      </a>
      <a
        aria-current={false}
        aria-disabled={false}
        className="item"
        onClick={[Function]}
        onKeyDown={[Function]}
        tabIndex={0}
        type="pageItem"
        value={3}
      >
        3
      </a>
      <a
        aria-current={false}
        aria-disabled={false}
        className="item"
        onClick={[Function]}
        onKeyDown={[Function]}
        tabIndex={0}
        type="pageItem"
        value={4}
      >
        4
      </a>
      <a
        aria-current={false}
        aria-disabled={false}
        className="item"
        onClick={[Function]}
        onKeyDown={[Function]}
        tabIndex={0}
        type="pageItem"
        value={5}
      >
        5
      </a>
      <a
        aria-current={false}
        aria-disabled={false}
        aria-label="Next item"
        className="item"
        onClick={[Function]}
        onKeyDown={[Function]}
        tabIndex={0}
        type="nextItem"
        value={2}
      >
        ⟩
      </a>
      <a
        aria-current={false}
        aria-disabled={false}
        aria-label="Last item"
        className="item"
        onClick={[Function]}
        onKeyDown={[Function]}
        tabIndex={0}
        type="lastItem"
        value={5}
      >
        »
      </a>
    </div>
  </div>
</div>
`;

exports[`PostListPagination component match snapshot wide screen /w props 2/3 1`] = `
<div
  className="three column row"
>
  <div
    className="right floated column"
  >
    <div
      aria-label="Pagination Navigation"
      className="ui pagination menu"
      role="navigation"
    >
      <a
        aria-current={false}
        aria-disabled={false}
        aria-label="First item"
        className="item"
        onClick={[Function]}
        onKeyDown={[Function]}
        tabIndex={0}
        type="firstItem"
        value={1}
      >
        «
      </a>
      <a
        aria-current={false}
        aria-disabled={false}
        aria-label="Previous item"
        className="item"
        onClick={[Function]}
        onKeyDown={[Function]}
        tabIndex={0}
        type="prevItem"
        value={2}
      >
        ⟨
      </a>
      <a
        aria-current={false}
        aria-disabled={false}
        className="item"
        onClick={[Function]}
        onKeyDown={[Function]}
        tabIndex={0}
        type="pageItem"
        value={1}
      >
        1
      </a>
      <a
        aria-current={false}
        aria-disabled={false}
        className="item"
        onClick={[Function]}
        onKeyDown={[Function]}
        tabIndex={0}
        type="pageItem"
        value={2}
      >
        2
      </a>
      <a
        aria-current={true}
        aria-disabled={false}
        className="active item"
        onClick={[Function]}
        onKeyDown={[Function]}
        tabIndex={0}
        type="pageItem"
        value={3}
      >
        3
      </a>
      <a
        aria-current={false}
        aria-disabled={false}
        className="item"
        onClick={[Function]}
        onKeyDown={[Function]}
        tabIndex={0}
        type="pageItem"
        value={4}
      >
        4
      </a>
      <a
        aria-current={false}
        aria-disabled={false}
        className="item"
        onClick={[Function]}
        onKeyDown={[Function]}
        tabIndex={0}
        type="pageItem"
        value={5}
      >
        5
      </a>
      <a
        aria-current={false}
        aria-disabled={false}
        aria-label="Next item"
        className="item"
        onClick={[Function]}
        onKeyDown={[Function]}
        tabIndex={0}
        type="nextItem"
        value={4}
      >
        ⟩
      </a>
      <a
        aria-current={false}
        aria-disabled={false}
        aria-label="Last item"
        className="item"
        onClick={[Function]}
        onKeyDown={[Function]}
        tabIndex={0}
        type="lastItem"
        value={5}
      >
        »
      </a>
    </div>
  </div>
</div>
`;

exports[`PostListPagination component match snapshot wide screen /w props 3/3 1`] = `
<div
  className="three column row"
>
  <div
    className="right floated column"
  >
    <div
      aria-label="Pagination Navigation"
      className="ui pagination menu"
      role="navigation"
    >
      <a
        aria-current={false}
        aria-disabled={false}
        aria-label="First item"
        className="item"
        onClick={[Function]}
        onKeyDown={[Function]}
        tabIndex={0}
        type="firstItem"
        value={1}
      >
        «
      </a>
      <a
        aria-current={false}
        aria-disabled={false}
        aria-label="Previous item"
        className="item"
        onClick={[Function]}
        onKeyDown={[Function]}
        tabIndex={0}
        type="prevItem"
        value={9}
      >
        ⟨
      </a>
      <a
        aria-current={false}
        aria-disabled={false}
        className="item"
        onClick={[Function]}
        onKeyDown={[Function]}
        tabIndex={0}
        type="pageItem"
        value={1}
      >
        1
      </a>
      <a
        aria-current={false}
        aria-disabled={true}
        className="item"
        onClick={[Function]}
        onKeyDown={[Function]}
        tabIndex={-1}
        type="ellipsisItem"
        value={5}
      >
        ...
      </a>
      <a
        aria-current={false}
        aria-disabled={false}
        className="item"
        onClick={[Function]}
        onKeyDown={[Function]}
        tabIndex={0}
        type="pageItem"
        value={6}
      >
        6
      </a>
      <a
        aria-current={false}
        aria-disabled={false}
        className="item"
        onClick={[Function]}
        onKeyDown={[Function]}
        tabIndex={0}
        type="pageItem"
        value={7}
      >
        7
      </a>
      <a
        aria-current={false}
        aria-disabled={false}
        className="item"
        onClick={[Function]}
        onKeyDown={[Function]}
        tabIndex={0}
        type="pageItem"
        value={8}
      >
        8
      </a>
      <a
        aria-current={false}
        aria-disabled={false}
        className="item"
        onClick={[Function]}
        onKeyDown={[Function]}
        tabIndex={0}
        type="pageItem"
        value={9}
      >
        9
      </a>
      <a
        aria-current={true}
        aria-disabled={false}
        className="active item"
        onClick={[Function]}
        onKeyDown={[Function]}
        tabIndex={0}
        type="pageItem"
        value={10}
      >
        10
      </a>
      <a
        aria-current={false}
        aria-disabled={false}
        aria-label="Next item"
        className="item"
        onClick={[Function]}
        onKeyDown={[Function]}
        tabIndex={0}
        type="nextItem"
        value={10}
      >
        ⟩
      </a>
      <a
        aria-current={false}
        aria-disabled={false}
        aria-label="Last item"
        className="item"
        onClick={[Function]}
        onKeyDown={[Function]}
        tabIndex={0}
        type="lastItem"
        value={10}
      >
        »
      </a>
    </div>
  </div>
</div>
`;
